src/metabase/api/common.clj

```diff
@@ -93,6 +95,21 @@
   []
   (check-403 *is-superuser?*))
 
+(defn check-has-general-permission
+  "If `advanced-permissions` is enabled, check `*current-user*` has general permission of type `perm-type`.
+  Set `require-superuser?` to `true` to perform a superuser check when `advanced-permissions` is disabled."
+  ([perm-type]
+   (check-has-general-permission perm-type true))
+
+  ([perm-type require-superuser?]
+   (classloader/require 'metabase.public-settings.premium-features
+                        'metabase.models.permissions)
+   (if ((resolve 'metabase.public-settings.premium-features/enable-advanced-permissions?))
+     (check-403 (or *is-superuser?*
+                    ((resolve 'metabase.models.permissions/set-has-general-permission-of-type)
+                     @*current-user-permissions-set* perm-type)))
```

Comment:
I think it would be slightly preferable to have a bit more of the permission checking logic in an EE namespace

ie:
1. Check if the EE ns can be resolved
2. If yes, call into an EE function to do the full permissions checking
3. Else, just check superuser if necessary